src/gwt/src/org/rstudio/studio/client/common/dependencies/DependencyManager.java

```diff
@@ -304,6 +304,39 @@ public void execute(Boolean succeeded)
       return deps.toArray(new Dependency[deps.size()]);
    }
    
+   public void withDataImportSAV(String userAction, final Command command)
+   {
+     withDependencies(
+        "Preparing Import from SAV",
+        userAction, 
+        dataImportSavDependenciesArray(), 
+        false,
+        new CommandWithArg<Boolean>()
+        {
+           @Override
+           public void execute(Boolean succeeded)
+           {
+              if (succeeded)
+                 command.execute();
+           }
+        }
+     );
+   }
+   
+   private ArrayList<Dependency> dataImportSavDependencies()
+   {
+      ArrayList<Dependency> deps = new ArrayList<Dependency>();
+      deps.add(Dependency.cranPackage("haven", "0.2.0"));
```

Comment:
Good point, added checks and marking disabled when the R requirement is not met.